zulipterminal/core.py

```diff
@@ -617,11 +617,18 @@ def narrow_to_all_mentions(self) -> None:
         self._narrow_to(anchor=None, mentioned=True)
 
     def deregister_client(self) -> None:
-        queue_id = self.model.queue_id
-        self.client.deregister(queue_id, 1.0)
+        raise urwid.ExitMainLoop
 
     def exit_handler(self, signum: int, frame: Any) -> None:
-        self.deregister_client()
+        question = urwid.Text(
+            ("bold", "Do you want to deregister the client?"),
+            "center",
+        )
+        popup_view = PopUpConfirmationView(
+            self, question, self.deregister_client, location="center"
+        )
+        self.loop.widget = popup_view
+        self.loop.run()
         sys.exit(0)
```

Comment:
I was surprised that we need the `loop.run()` here, but perhaps the signal handling interrupts the loop in some way. It seems to behave strangely without it?

Reply:
Thank you for your comment and for reviewing the code. I appreciate your feedback.
The absence of `loop.run()` results in the popup not appearing, it appears that you are correct, the signal handling is interrupting the loop in some way.